src/poetry/installation/chooser.py

```diff
@@ -103,6 +104,15 @@ def _get_links(self, package: Package) -> list[Link]:
                 selected_links.append(link)
                 continue
 
+            if (
+                link.hash_name
+                and not link.hash_name.startswith("sha")
+                and isinstance(repository, PyPiRepository)
```

Comment:
Should we consider makign this apply broadly?

Reply:
Broadly as in for any repo type? The `get_sha_method_from_link` method would need to be added to `Repository` in that case since this check only exists to ensure the existence of the required method. I'm not entirely sure what the logic of the method would need to be for all the other types.